src/widgets/_windows.scss

```diff
@@ -1,71 +1,99 @@
-window:not(.popup),
+window.csd:not(.popup):not(.menu),
 dialog {
     border-radius: 0 0 rem(6px) rem(6px);
 
     > decoration {
         border-radius: rem(6px);
         margin: rem(12px);
-
-        &:backdrop {
-            box-shadow: shadow(2);
-        }
     }
 
     > .titlebar {
         border-radius: rem(6px) rem(6px) 0 0;
     }
 }
 
-window:not(.popup) {
-    // Intentionally not in ems since it's used as a stroke
-    box-shadow: 0 0 0 1px $toplevel-border-color;
-
-    > decoration {
-        box-shadow: shadow(4);
-
-        &:backdrop {
-            box-shadow: shadow(2)
-        }
-    }
+window:not(.popup):not(.menu) {
+    // We only draw shadows for client-side decorations
+    &.csd {
+        // Intentionally not in ems since it's used as a stroke
+        box-shadow: 0 0 0 1px $toplevel-border-color;
 
-    &.tiled {
         > decoration {
-            box-shadow: shadow(1);
+            box-shadow: shadow(4);
+
+            &:backdrop {
+                box-shadow:
+                    // Workaround for a resizing issue
```

Comment:
@danrabbit can you be more specific here and/or link to an issue? It would be good to be able to better understand _what_ it's working around as well by reading the comment, if possible